src/cpl.rs

```diff
@@ -155,6 +166,47 @@ impl Debug for CslStringList {
     }
 }
 
+/// Convenience shorthand for specifying an empty `CslStringList` to functions accepting
+/// `Into<CslStringList>`.
+///
+/// # Example
+///
+/// ```rust, no_run
+/// use gdal::cpl::CslStringList;
+/// fn count_opts<O: Into<CslStringList>>(opts: O) -> usize {
+///     opts.into().len()
+/// }
+///
+/// assert_eq!(count_opts(()), 0);
+/// ```
+impl From<()> for CslStringList {
+    fn from(_: ()) -> Self {
+        CslStringList::default()
+    }
+}
+
+/// Convenience for creating a [`CslStringList`] from a slice of _key_/_value_ tuples.
+///
+/// # Example
+///
+/// ```rust, no_run
+/// use gdal::cpl::CslStringList;
+/// fn count_opts<O: Into<CslStringList>>(opts: O) -> usize {
+///     opts.into().len()
+/// }
+///
+/// assert_eq!(count_opts(&[("One", "1"), ("Two", "2"), ("Three", "3")]), 3);
+/// ```
+impl<const N: usize> From<&[(&str, &str); N]> for CslStringList {
+    fn from(pairs: &[(&str, &str); N]) -> Self {
+        let mut result = Self::default();
+        for (k, v) in pairs {
+            result.set_name_value(k, v).expect("valid key/value pair");
```

Comment:
```suggestion
            result.set_name_value(k, v).expect("invalid key/value pair");
```

I think the message would be something like `valid key/value pair: "Invalid characters in name: '#'"`, which doesn't make sense.

And going back a little, I'm not sure what to say about this impl. `from` should be infallible, so `try_from` would be better, which this clearly isn't. On the other hand, it's probably going to be used with compile-constant keys and values. Is that why you've implemented it for fixed-size arrays (as opposed to `&[(&str, &str)]`)?

Reply:
> Is that why you've implemented it for fixed-size arrays

No. More of a result of me playing type golf; Without the `const` parameter I get this error. Probably me being dumb.

<img width="1032" alt="image" src="https://user-images.githubusercontent.com/131013/211640561-3d2453be-8c66-446e-90c2-1107e98d786c.png">


The fact that `set_name_value` is fallible is really annoying, but understandable. Going with `try_from` pretty much eliminates the benefits of having this. So if you're in doubt about it, I think I should just remove it all together.

Reply:
Oof, yes, I think you need to cast it, like in https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0c8e96e6b31a1d7ac28a76cce7abbd1b. Anyway, I suppose it's fine.

Reply:
> I think the message would be something like `valid key/value pair: "Invalid characters in name: '#'"`, which doesn't make sense.


Are you 100% sure? (I'm not) 😄 

Per [the docs](https://doc.rust-lang.org/core/result/enum.Result.html#recommended-message-style)

> We recommend that expect messages are used to describe the reason you expect the Result should be Ok.

Reply:
Can you test? 😀 I can't right now, sorry.

My reading of the docs is something like `CslStringList::try_from(&["FOO", "BAR"]).expect("obviously correct")`, that is, user code. It's different in library code which can and will actually panic.

Reply:
I'm not sure I'm following your line of thinking here. Happy to submit a new PR revisiting this.

Reply:
Indeed, this produces:

```
thread 'cpl::tests::from_impl' panicked at 'valid key/value pair: BadArgument("Invalid characters in name: 'TWO#'")', src/cpl.rs:204:41
```

compared to a random example from `libcore`:

```
self.checked_add(rhs).expect("overflow when adding durations")
```